src/Npgsql/NpgsqlParameter.cs

```diff
@@ -536,17 +536,19 @@ public override void ResetDbType()
         /// is a copy of the current instance.
         /// </summary>
         /// <returns>A new <see cref="NpgsqlParameter">NpgsqlParameter</see> that is a copy of this instance.</returns>
-        public NpgsqlParameter Clone()
-        {
+        public NpgsqlParameter Clone() => CloneCore();
```

Comment:
Any reason to separate out a CloneCore, rather than to just make Clone virtual?

Reply:
It's possible to inherit from `NpgsqlParameter`, and I don't want to increase public surface more than it's now. Even more I would like to remove all `Clone` methods as described in #3422.